spec/models/condition_spec.rb

```diff
@@ -64,6 +64,78 @@
     end
   end
 
+  describe ".create_and_update_form!" do
+    let(:form) { create(:form) }
+    let(:routing_page) { create(:page, form:) }
+    let(:goto_page) { create(:page, form:) }
+    let(:condition_params) do
+      { check_page_id: routing_page.id,
+        routing_page_id: routing_page.id,
+        answer_value: "Yes",
+        goto_page_id: goto_page.id,
+        skip_to_end: false,
+        exit_page_heading: nil,
+        exit_page_markdown: nil }
+    end
+
+    it "saves the condition to the database" do
+      expect {
+        described_class.create_and_update_form!(**condition_params)
+      }.to change(described_class, :count).by(1)
+    end
+
+    it "creates the condition with the given attributes" do
+      created_condition = described_class.create_and_update_form!(**condition_params)
+      expect(created_condition.check_page_id).to eq(routing_page.id)
+      expect(created_condition.routing_page_id).to eq(routing_page.id)
+      expect(created_condition.goto_page_id).to eq(goto_page.id)
+      expect(created_condition.answer_value).to eq("Yes")
+      expect(created_condition.skip_to_end).to be(false)
```

Comment:
It might be simpler to use the `have_attributes` [[1](https://rubydoc.info/gems/rspec-expectations/3.13.5/RSpec/Matchers#have_attributes-instance_method)] matcher?

```suggestion
      expect(created_condition).to have_attributes(
        check_page_id: routing_page.id,
        routing_page_id: routing_page.id,
        goto_page_id: goto_page.id,
        answer_value: "Yes",
        skip_to_end: false,
      )
```